app/controllers/hubstats/deploys_controller.rb

```diff
@@ -75,5 +58,32 @@ def create
         end
       end
     end
+
+    def valid_repo(repo)
+      if !repo.empty?
+        @deploy.repo_id = repo.first.id.to_i
```

Comment:
Having hte assignment in here is a bit misleading.  I like the validity function, but consider moving the assignment back under the if statement.